### ISSUE TITLE
Can't Remove Stream Data from DNN Search Index using CustomSearch function

### ISSUE THREAD

**Issue by tpperlman** [2sic/2sxc#2003]
**I'm submitting a ...** 
[x] bug report

**...about**
[x] Razor templating
[x] DNN parts

**Current behavior**
2sxc content from Razor templates are property indexed by DNN, but we are not able to exclude specific content from the index. I based my code on the solution provided in the stack overflow article at [https://stackoverflow.com/questions/43387615/2sxc-how-to-disable-searching-indexing](https://stackoverflow.com/questions/43387615/2sxc-how-to-disable-searching-indexing)

**Expected behavior**
Using the CustomSearch function and setting the searchInfos["StreamName"] object to an empty list should remove all the items.

**Minimal reproduction of the problem with instructions**
Create an app, create data and add some items, create a query, create a view, create a razor template for the view. In the view, add the following c# code:

```
@functions
{
    public override void CustomizeSearch(Dictionary<string, List<ToSic.SexyContent.Search.ISearchInfo>> searchInfos, DotNetNuke.Entities.Modules.ModuleInfo moduleInfo, DateTime startDate)
    {
        searchInfos["Default"] = new List<ToSic.SexyContent.Search.ISearchInfo>();
    }
}
```
Re-index all site content, run the scheduled task for the index, and then do a search for some of the data you entered for the app. Notice that the content still appears.

**What is the motivation / use case for changing the behavior?**
We may want to restrict some content from being displayed to users in the search results.

**Please tell us about your environment:**
IIS

* **2sxc version(s):** 10.26
* **Browser:** all
* **DNN:** 09.05.00
* **Language:** all

**Anything you would like to add**

**Comment by iJungleboy** [comment]
I'm not sure if you're running into a bug. He're what you should know:

1. DNN includes the timestamp of last-indexing to only re-index changed or new content
1. so if you didn't clear the index-cache, only changed stuff is re-indexed

My guess is, that you didn't clear your search-index (in the portal settings). Could you verify?

**Comment by tpperlman** [comment]
Hi Daniel, thanks for the quick response. My bad for not mentioning it earlier, but I did do a rebuild of the site catalog and then triggered the scheduled task. I even went as far as deleting all the Lucene search files in the App_Data folder too and rebuilding from scratch. Each time the site was re-indexed, the entries continue to appear.

I also thought that the "Allow Index" setting in module settings might block these items from being indexed too, but that doesn't work either. As a matter of fact, each time I unchecked the "Allow Index" option in module settings and saved it, the option remained checked. I even went as far as manually inserting the setting into the ModuleSettings table and this too had no affect on the indexing. IMHO, the "Allow Indexing" option should be the preferred method to exclude content from indexing since it is built into DNN.

**Comment by iJungleboy** [comment]
We've just been doing some testing with the search index and we didn't see the same problem. 
Please note that in case you have other streams containing the same item, those streams too will be indexed. So you may want to remove certain streams you don't want for the indexer. 

Something like this:

```
@functions{
  /// <summary>
  /// Populate the search - ensure that each entity has an own url/page
  /// </summary>
  public override void CustomizeSearch(Dictionary<string, List<ISearchItem>> searchInfos, IContainer moduleInfo, DateTime beginDate) {
    var parts = CreateInstance("shared/_Parts.cshtml");
    foreach (var si in searchInfos["SearchIndex"]) {
      var article = AsDynamic(si.Entity);
      si.QueryString = parts.DetailPageQueryString(article, "");
    }

    // Remove not needed streams
    var keys = searchInfos.Keys.ToList();
    foreach(var key in keys) {
      if (key != "SearchIndex") {
        searchInfos.Remove(key);
      }
    }
  }
}
```

**Comment by iJungleboy** [comment]
Will close because of inactivity and re-open if something still pops up

**Comment by tpperlman** [comment]
I have created a new stand-alone DNN installation that demonstrates the issue. I can give you permission to do some further research, but the issue persists.

I will email you directly with a host login to the site for you to look into it.

**Comment by tpperlman** [comment]
Have you had a chance to review the email I sent? I see the issue is still closed. I know you have been busy with a new release.

**Comment by iJungleboy** [comment]
Sorry, didn't have a chance yet - and forgot to re-open.

**Comment by iJungleboy** [comment]
@tpperlman I finally got to looking into it again. 

It seems that there is a bug in 2sxc related to Razor - so if it's using the "old" Razor model (without `@inherits ToSic.Sxc.Dnn.RazorComponent`) it doesn't seem to work correctly. 

Once I changed it to a new RazorComponent it works. So this is a bug. Check out the changes I made. 

So for now you should be able to get it to work, but it's not fixed in 2sxc - just upgrade the relevant views to RazorComponents.

**Comment by iJungleboy** [comment]
ok, just fixed it :) will be in 11.03

**Comment by tpperlman** [comment]
You rock, thank you Daniel. Keep healthy!
